zeppelin-zengine/src/main/java/org/apache/zeppelin/storage/LocalConfigStorage.java

```diff
@@ -97,12 +107,22 @@ public void saveCredentials(String credentials) throws IOException {
     writeToFile(credentials, credentialPath);
   }
 
-  private String readFromFile(File file) throws IOException {
-    return IOUtils.toString(new FileInputStream(file));
+  private String readFromFile(Path path) throws IOException {
+    return IOUtils.toString(new FileInputStream(path.toFile()));
   }
 
-  private void writeToFile(String content, File file) throws IOException {
-    FileOutputStream out = new FileOutputStream(file);
+  private void writeToFile(String content, Path path) throws IOException {
+    if (!Files.exists(path)) {
+      Files.createFile(path);
+      try {
+        Set<PosixFilePermission> permissions = EnumSet.of(OWNER_READ, OWNER_WRITE);
+        Files.setPosixFilePermissions(path, permissions);
```

Comment:
should it change the permission even if it exists? it might help older version migration.
also would it make sense to only set permission when failing to write? I realize that increases the complicity but might be more compatible?